dtfs-central-api/src/errors/transaction-failed.error.ts

```diff
@@ -0,0 +1,11 @@
+import { HttpStatusCode } from 'axios';
+import { ApiError } from '.';
+
+export class TransactionFailedError extends ApiError {
+  constructor(message: string = 'Transaction failed') {
+    super({
+      message,
+      status: HttpStatusCode.InternalServerError,
```

Comment:
Because the transaction could have failed for a variety of reasons, I think a `500` is appropriate here as, hopefully, any issues related to the shape of the request have already been validated before the transaction starts. Any thoughts?